trading_cost.py

```diff
@@ -0,0 +1,64 @@
+"""
+A file to build cost objects.
+"""
+
+from collections import defaultdict
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.instruments = instruments
+        self.quantity = defaultdict(int)
+        self.value = defaultdict(int)
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price, liquidation):
+        if liquidation:
+            for instrument in self.instruments:
+                portfolio[-1][instrument] = (
+                    0 if instrument is not self.base_ins else 1)
+        return self.calculate(portfolio[-1], price)
+
+    def calculate(self, today, price):
+        """
+        Calculate pnl and produce cost.
+
+        We assume that our investing money is maintained in 1.
+        That make it more general to evaluate the perfoormance for the alpha.
+        For detailed, please read our wiki pages.
+        """
+        pnl = {
+            instrument : 0 for instrument in self.instruments}
+        cost = {
+            instrument : 0 for instrument in self.instruments}
```

Comment:
In my thinking, the portfolio we get may not have all instrument we traded. for example:
We will trade 'USD', 'JPY', 'EUR'
portfolio for 10/01 only have 'USD', 'JPY'
portfolio for 10/02 only have 'USD', 'EUR'.
Thus, if we use defaultdict in this part, we could not get all the key (in line 39, we only get the key in portfolio.)
Because of that, I use list comprehensive for this two args.

Reply:
Fair point. Thanks!